corehq/apps/aggregate_ucrs/views.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib import messages
+from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext_lazy
+
+from corehq import toggles
+from corehq.apps.aggregate_ucrs.models import AggregateTableDefinition
+from corehq.apps.aggregate_ucrs.sql.adapter import AggregateIndicatorSqlAdapter
+from corehq.apps.aggregate_ucrs.tasks import populate_aggregate_table_data_task
+from corehq.apps.domain.decorators import login_and_domain_required, login_or_basic
+from corehq.apps.userreports.views import BaseUserConfigReportsView, swallow_programming_errors, \
+    export_sql_adapter_view
+from corehq.apps.users.decorators import require_permission
+from corehq.apps.users.models import Permissions
+
+
+class BaseAggregateUCRView(BaseUserConfigReportsView):
+
+    @method_decorator(toggles.AGGREGATE_UCRS.required_decorator())
```

Comment:
You can put this on the class by specifying `name='dispatch'` and then you don't need to override the method.

Reply:
neat trick